### PR TITLE
issue 465.2.2: remove redudnant heap allocations and simplify API

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

**Comment by vasiliy-arangodb** [comment]
https://github.com/arangodb/enterprise/pull/139

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/947/

File: arangod/V8Server/v8-collection.cpp [hunk 1]
```diff
@@ -72,6 +72,40 @@
...
+        ? ci->getCollection(vocbase.name(), TRI_ObjectToString(val)) : nullptr;
+    } catch (...) {
+      // NOOP
+    }
+
+    return nullptr; // not found
+  }
+
+  // number
+  if (val->IsNumber() || val->IsNumberObject()) {
```
Comment (graetzer) [inline arangod/V8Server/v8-collection.cpp]: don't you have to support lookups by plan ID in the coordinator as well ?

File: arangod/V8Server/v8-collection.cpp [hunk 2]
```diff
@@ -191,15 +227,12 @@ static int ParseDocumentOrDocumentHandle(v8::Isolate* isolate,
     if (ServerState::instance()->isCoordinator()) {
       ClusterInfo* ci = ClusterInfo::instance();
       try {
-        std::shared_ptr<LogicalCollection> col =
-            ci->getCollection(vocbase->name(), collectionName);
-        auto colCopy = col->clone();
-        collection = colCopy.release();
+        collection = ci->getCollection(vocbase->name(), collectionName).get();
```
Comment (graetzer) [inline arangod/V8Server/v8-collection.cpp]: this seems a bit dangerous, shouldn't you be keeping the shared_ptr copy ?

File: arangod/V8Server/v8-collection.cpp (hunk 1 above)
Comment (vasiliy-arangodb) [reply arangod/V8Server/v8-collection.cpp]: this was moved from line 1977 and combined with lines 2268/2351

File: arangod/V8Server/v8-collection.cpp (hunk 2 above)
Comment (vasiliy-arangodb) [reply arangod/V8Server/v8-collection.cpp]: done

**Pull request merged**